test/new-e2e/tests/agent-platform/ddot/ddot_install_test.go

```diff
@@ -142,6 +142,10 @@ func (is *ddotInstallSuite) ConfigureAndRunAgentService(VMclient *common.TestCli
 		} else {
 			ExecuteWithoutError(t, VMclient, "sudo systemctl restart datadog-agent.service")
 		}
+
+		// Reset systemd failure state and start ddot service
+		ExecuteWithoutError(t, VMclient, "sudo systemctl reset-failed datadog-agent-ddot.service")
+		ExecuteWithoutError(t, VMclient, "sudo systemctl start datadog-agent-ddot.service")
```

Comment:
This test should be passing without these two lines, the expected behavior is that DDOT will start successfully when `datadog-agent.service` is restarted. What errors were you observing before adding these lines?

Also, you should remove this line since you added support for it in the postinst:
```
ExecuteWithoutError(t, VMclient, "sudo sh -c \"printf 'otelcollector:\\n enabled: true\\n' >> /etc/datadog-agent/datadog.yaml\"")
```